MaxText/maxengine.py

```diff
@@ -298,6 +298,99 @@ def prefill_aot(
 
     return self.prefill(params=params, padded_tokens=padded_tokens, true_length=true_length, rng=rng)
 
+  @functools.partial(jax.jit, static_argnums=(0,), static_argnames=("num_samples",))
+  def prefill_multisampling(
+      self,
+      *,
+      params: Params,
+      existing_prefix: Optional[jax.Array] = None,
+      padded_tokens: jax.Array,
+      true_length: int,
+      sampler: Optional[Callable[[Any], Any]] = None,  # pylint: disable=unused-argument
+      rng: Optional[PRNGKeyType] = None,
+      num_samples: int = 1,
+  ) -> Tuple[Prefix, engine_api.ResultTokens]:
+    """Computes a kv-cache for a new generate request.
+
+    With multi-sampling, the engine will generate multiple first tokens in the
+    prefilling stage. The number of tokens is specified by num_samples.
+    """
+    if existing_prefix:
+      raise ValueError("We don't know what to do with existing_prefix")
```

Comment:
nit: should we update it this is not supported or something similar?

Reply:
Will remove it along with the one in prefill()

Reply:
Can you add a TODO if you don't want to update it now?